contracts/extensions/VotingReputation.sol

```diff
@@ -0,0 +1,817 @@
+
+    delete poll.events;
+
+    poll.events[CREATE] = now;
+    poll.domainId = _newDomainId;
+    poll.skillId = newDomainSkillId;
+    poll.skillRep = checkReputation(_pollId, address(0x0), _key, _value, _branchMask, _siblings);
+  }
+
+  function executePoll(uint256 _pollId) public {
+    Poll storage poll = polls[_pollId];
+    PollState pollState = getPollState(_pollId);
+
+    require(pollState != PollState.Failed, "voting-rep-poll-failed");
+    require(pollState != PollState.Closed, "voting-rep-poll-escalation-window-open");
+    require(pollState != PollState.Executed, "voting-rep-poll-already-executed");
+    require(pollState == PollState.Executable, "voting-rep-poll-not-executable");
+
+    poll.executed = true;
+
+    bool canExecute = (
+      poll.stakes[NAY] <= poll.stakes[YAY] &&
+      poll.votes[NAY] <= poll.votes[YAY]
+    );
+
+    if (getSig(poll.action) == CHANGE_FUNCTION) {
+      bytes32 structHash = hashExpenditureStruct(poll.action);
+      expenditurePollCounts[structHash] = sub(expenditurePollCounts[structHash], 1);
+
+      // Release the claimDelay if this is the last active poll
+      if (expenditurePollCounts[structHash] == 0) {
+        bytes memory claimDelayAction = createClaimDelayAction(poll.action, 0);
+        executeCall(_pollId, claimDelayAction);
+      }
+
+      // Conditions:
+      //  - Yay side staked and nay side did not, and doman has sufficient vote power
+      //  - Both sides staked and yay side won, with sufficient vote power
+
+      uint256 votePower;
+      if (poll.stakes[NAY] < getRequiredStake(_pollId)) {
+        votePower = wmul(poll.skillRep, votePowerFraction);
+      } else {
+        votePower = poll.votes[YAY];
+      }
+
+      bytes32 slotHash = hashExpenditureSlot(poll.action);
+      if (expenditurePastPolls[slotHash] < votePower) {
+        expenditurePastPolls[slotHash] = votePower;
+        canExecute = canExecute && true;
+      } else {
+        canExecute = canExecute && false;
+      }
+    }
+
+    bool success;
+    if (canExecute) {
+      success = executeCall(_pollId, poll.action);
+    }
+
+    emit PollExecuted(_pollId, poll.action, success);
+  }
+
+  function claimReward(
+    uint256 _pollId,
+    uint256 _permissionDomainId, // For extension's arbitration permission
+    uint256 _childSkillIndex, // For extension's arbitration permission
+    address _user,
+    uint256 _vote
+  )
+    public
+  {
+    Poll storage poll = polls[_pollId];
+    require(
+      getPollState(_pollId) == PollState.Executed ||
+      getPollState(_pollId) == PollState.Failed,
+      "voting-rep-not-failed-or-executed"
+    );
+
+    // Calculate how much of the stake is left after voter compensation (>= 90%)
+    uint256 stake = stakes[_pollId][_user][_vote];
+    uint256 totalStake = add(poll.stakes[NAY], poll.stakes[YAY]);
+    uint256 rewardFraction = wdiv(poll.unpaidRewards, totalStake);
+    uint256 rewardStake = wmul(stake, rewardFraction);
```

Comment:
I'd like to punt on this for now -- we've been going back-and-forth about this and there are argument for both sides (i.e. why should there be an abrupt discontinuity in reward based on a potentially small vote?). If it becomes a problem in practice we can update the behavior later.

Reply:
If there are arguments for both sides, why isn't the 'default' position the one in the whitepaper?

We seem to be habitually increasing the cost of being right. There was already a loss in liquidity (and always has been), and during this review process we have already decided that it's too hard to prevent the loss of control of the tokens during a vote (and therefore leading to potential loss of colony payouts if they occur while a dispute is happening). Now, you're saying that even if I'm right, I might not get all of my tokens back? It's becoming less and less appealing to stake against a decision that I think should not happen, even if I think I'm right, which seems something that we shouldn't be taking likely.